src/cluster_legacy.c

```diff
@@ -1252,6 +1252,57 @@ void clusterInitLast(void) {
 
 /* Called when a cluster node receives SHUTDOWN. */
 void clusterHandleServerShutdown(void) {
+    if (nodeIsPrimary(myself) && server.auto_failover_on_shutdown) {
+        /* Find the first best replica, that is, the replica with the largest offset. */
+        int old_replica = 0;
```

Comment:
nit. legacy? "old" struck me as being an "out of sync" replica at first.

```suggestion
        int legacy_replica = 0;
```